src/openassetio-core/TraitsData.cpp

```diff
@@ -65,6 +65,16 @@ class TraitsData::Impl {
   PropertiesByTrait data_;
 };
 
+TraitsDataPtr TraitsData::make() { return std::shared_ptr<TraitsData>(new TraitsData()); }
+
+TraitsDataPtr TraitsData::make(const TraitSet& traitSet) {
+  return std::shared_ptr<TraitsData>(new TraitsData(traitSet));
+}
+
+TraitsDataPtr TraitsData::make(const TraitsDataPtr& other) {
```

Comment:
[consider] / [plan] This seems like a good place to use a `TraitsDataConstPtr` as the argument. Would also be nice to use in the tests where we take a copy of the data for comparisons.  

I'm not sure if taking a ConstPtr and passing a non-const Ptr would cause a ConstPtr temporary to be created and hence do a pointless refcount.  But it still feels like the right thing to do.

Reply:
Good call, Ill fix that.

I _think_ we use it in the tests already as we bind this method to python for the copy constructor?